sips/sip-010/sip-010-fungible-token-standard.md

```diff
@@ -0,0 +1,172 @@
+# Preamble
+
+Sip Number: 010
+
+Title: Standard Trait Definition for Non-Fungibe Tokens
+
+Author: Hank Stoever (hstove@gmail.com)
+
+Consideration: Technical
+
+Type: Standard
+
+Status: Draft
+
+Created: 25 January 2021
+
+License: CC0-1.0
+
+Sign-off:
+
+Layer: Traits
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+Fungible tokens are digital assets that can be sent, received, combined, and divided. Most forms of currency and cryptocurrencies are fungible tokens. They have become a building block of almost all blockchains. This SIP aims to provide a flexible and easy-to-implement standard that can be used by developers on the Stacks blockchain when creating their own tokens.
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0 Universal license, available at https://creativecommons.org/publicdomain/zero/1.0/
+This SIP’s copyright is held by the Stacks Open Internet Foundation.
+
+# Introduction
+
+Digital assets can be divided into two categories, based on the token's fungibility. A _fungible_ token can be broken down into small units and added together. An owner of a fungible asset only needs to care about their balance, that is, the total amount of a particular fungible asset that they own. Most well-known currencies are fungible.
+
+For example, if a user owns 10 units of a fungible asset, they may send 2 units to a different user. At this point, their balance is 8 units. If they later receive more units, their total balance will be updated. For fungible tokens, there is no difference between any two different amounts of the fungible token.
+
+On blockchains, fungible tokens are a core component. Blockchains with smart contracts, including the Stacks blockchain, allow developers and users to create and interact with smart contracts that use fungible tokens.
+
+# Specification
+
+The fungible token trait, `ft-trait`, has a few functions:
+
+## Trait functions
+
+### Transfer
+
+`(transfer ((amount uint) (sender principal) (recipient principal)) (response bool uint))`
+
+Transfer the fungible token from the sender of this transaction to the recipient. The `amount` is an unsigned integer. It is recommended that implementing contracts use the built-in `ft-transfer` Clarity method. If the sender does not have enough tokens to complete the transaction, the transaction should abort and return an `(err uint)`.
+
+This method must be defined with `define-public`, as it alters state, and should be externally callable.
+
+Contract implementers should take note to perform authorization of the `transfer` method. For example, most fungible token contracts should enforce that the `sender` argument is equal to the `tx-sender` keyword in Clarity.
```

Comment:
Enforcing the sender to be the `tx-sender` implies that the token can't be managed by a smart contract as agent. This seems to be too restrictive and not desirable in most cases.

Better:  .. is equal  to the `tx-sender` or `contract-caller` keyword.

Reply:
I'm not sure this is true. If a smart contract needs to manage fungible tokens, it can use `as-contract` to modify the `tx-sender` variable.

Reply:
I think the issue here is that the wording is a little too strong -- it's suggesting that the caller really ought to ensure `(is-eq (tx-sender sender))`.  Maybe change the wording to something less forceful, such as this?

"For example, a fungible token contract that wants to make sure that only the transaction's sender is able to move the requested tokens might first check that the `sender` argument is equal to `tx-sender`."